### ISSUE TITLE
[Bug]: Android: new notifications replace previous ones, because they all have androiNotificationId equal to 0 (v5.0.0)

### ISSUE THREAD

**Issue by fmorau** [OneSignal/react-native-onesignal#1551]
### What happened?

Hello react-native-onesignal team,

I recently updated to react-native-onesignal v5 and I'm experiencing an issue on Android where push notifications are replacing each other, so I can only see the last one when I pull down the tray.

After some debugging, I found that the `osNotificationReceivedEvent.getNotification().getAndroidNotificationId()` method always returns 0, causing new notifications to replace previous ones because they have the same IDs.

I couldn't find any information about this issue online, so I wanted to report it here in case it's a bug in the library. Any help or guidance would be greatly appreciated.

Thank you!

### Steps to reproduce?

```Markdown
1. Install `react-native-onesignal` v5.0.0
2. Create `NotificationServiceExtension` for Android
3. Launch the app and subscribe to notifications (OneSignal.login + Notifications.OptIn())
4. Send push notifications to subscribed device via REST API from backend
```

### What did you expect to happen?

I expected that after updating to react-native-onesignal v5, push notifications on Android would continue to work as they did before the update. 
Specifically, I expected that multiple push notifications would be displayed in the notification tray without replacing each other, so that I could see all of them when I pull down the tray.

### React Native OneSignal SDK version

5.0.0

### Which platform(s) are affected?

- [ ] iOS
- [X] Android

### Relevant log output

```Shell
public void onNotificationReceived(INotificationReceivedEvent osNotificationReceivedEvent) {
  IDisplayableMutableNotification notification = osNotificationReceivedEvent.getNotification();
  int notificationId = notification.getAndroidNotificationId();
  System.out.print("notificationId: " + notificationId); // Always 0
}
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by jennantilla** [comment]
@fmorau thank you for reporting. We are currently investigating a fix for this and will let you know once it's been released.

**Comment by fmorau** [comment]
@jennantilla do you have any ETA for a new version with this fix?
We are currently in regression stage prior release and I need to take decision regarding it

**Comment by fmorau** [comment]
Hello, @jennantilla!
I would appreciate, if you have any updates for me :)

**Comment by jennantilla** [comment]
Hello @fmorau thank you for your patience! We are currently working on a fix for this, I'd estimate an ETA of late next week but hopefully sooner.

**Comment by fmorau** [comment]
@jennantilla great, thank you 🙏
We rely on you, guys 🙌

**Comment by Romick2005** [comment]
I have the same issue and can add, that sending push notifications from onesignal dashboard leads to the same behavior (no multiple push notification, but only latest one), even having unique Collapse ID.

**Comment by Romick2005** [comment]
I've fixed it by adding Android specific field "android_group" with unique value inside raw notification before sending it from backend to onesignal rest api

**Comment by fmorau** [comment]
Hello, @jennantilla! 
Could you please update us regarding current status of the issue, 
if everything is goind according to the plan and it should be fixed by the end of the week?
Thank you 🙏

**Comment by sakdog** [comment]
Same problem here. I downgraded back to 4.5.2 and now notification history displayed as expected.
Because of breaking changes in v5 if you go back to 4.5.2 make sure you modify your code with the functions of the previous version.
https://documentation.onesignal.com/v9.0/docs/react-native-sdk-setup

**Comment by jennantilla** [comment]
Hi everyone, thank you for your patience.

We have a fix for this in our latest release, [5.0.1](https://github.com/OneSignal/react-native-onesignal/releases/tag/5.0.1). Please let us know if you have any additional questions or concerns!
